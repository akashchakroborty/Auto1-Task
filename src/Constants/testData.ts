import { CarsResponse } from '../redux/types/carsTypes';

export const MOCK_CARS_DATA: CarsResponse = {
  cars: [
    {
      stockNumber: 97074,
      manufacturerName: 'Porsche',
      modelName: '718 Cayman',
      color: 'black',
      mileage: { number: 100056, unit: 'km' },
      fuelType: 'Diesel',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 12876,
      manufacturerName: 'Tesla',
      modelName: 'Roadster',
      color: 'yellow',
      mileage: { number: 100481, unit: 'km' },
      fuelType: 'Petrol',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 18029,
      manufacturerName: 'Fiat',
      modelName: '131',
      color: 'white',
      mileage: { number: 100710, unit: 'km' },
      fuelType: 'Petrol',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 65672,
      manufacturerName: 'Skoda',
      modelName: 'Felicia',
      color: 'black',
      mileage: { number: 100751, unit: 'km' },
      fuelType: 'Petrol',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 42578,
      manufacturerName: 'Mercedes-Benz',
      modelName: 'Sprinter',
      color: 'yellow',
      mileage: { number: 100782, unit: 'km' },
      fuelType: 'Diesel',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 70917,
      manufacturerName: 'Tesla',
      modelName: 'Roadster',
      color: 'green',
      mileage: { number: 100929, unit: 'km' },
      fuelType: 'Diesel',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 53029,
      manufacturerName: 'Tesla',
      modelName: 'Model X',
      color: 'black',
      mileage: { number: 100997, unit: 'km' },
      fuelType: 'Diesel',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 77489,
      manufacturerName: 'Chrysler',
      modelName: 'Viper',
      color: 'red',
      mileage: { number: 101001, unit: 'km' },
      fuelType: 'Petrol',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 98739,
      manufacturerName: 'Tesla',
      modelName: 'Roadster',
      color: 'blue',
      mileage: { number: 101066, unit: 'km' },
      fuelType: 'Petrol',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 72351,
      manufacturerName: 'Porsche',
      modelName: '968',
      color: 'green',
      mileage: { number: 101069, unit: 'km' },
      fuelType: 'Diesel',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
  ],
  totalPageCount: 100,
  totalCarsCount: 1000,
};

export const MOCK_COLORS = {
  colors: ['red', 'blue', 'green', 'black', 'yellow', 'white', 'silver'],
};

export const MOCK_MANUFACTURERS = {
  manufacturers: [
    {
      name: 'Audi',
      models: [
        {
          name: '100',
        },
      ],
    },
  ],
};
