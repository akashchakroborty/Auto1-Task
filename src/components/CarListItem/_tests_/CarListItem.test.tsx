import React from 'react';
import { shallow } from 'enzyme';
import { MOCK_CARS_DATA } from '../../../Constants/testData';
import { CarListItemProps, CarListItem, CarListItemSkeleton } from '..';

describe('should test the CarListItem Component', () => {
  const mockProps: CarListItemProps = {
    carItem: MOCK_CARS_DATA.cars[0],
  };
  test('should render the component', () => {
    const wrapper = shallow(<CarListItem {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('should test the CarListItemSkeleton Component', () => {
  test('should render the component', () => {
    const wrapper = shallow(<CarListItemSkeleton />);
    expect(wrapper).toMatchSnapshot();
  });
});
