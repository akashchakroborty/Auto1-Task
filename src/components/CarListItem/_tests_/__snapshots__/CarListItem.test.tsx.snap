// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the CarListItem Component should render the component 1`] = `
<ForwardRef(Card)
  className="makeStyles-card-1"
  elevation={0}
  raised={false}
  square={true}
  variant="outlined"
>
  <ForwardRef(CardMedia)
    alt="Car Thumbnail"
    component="img"
    height={75}
    image="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
    title="Car Thumbnail"
    width={100}
  />
  <ForwardRef(CardContent)
    className="makeStyles-content-2"
  >
    <ForwardRef(Typography)
      fontWeight={700}
      variant="h6"
    >
      Porsche
       
      718 Cayman
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      style={
        Object {
          "textTransform": "capitalize",
        }
      }
      variant="subtitle2"
    >
      Stock #
       
      97074
       
      -
       
      100,056
       
      KM
       
      -
       
      Diesel
       
      -
       
      black
    </ForwardRef(Typography)>
    <ForwardRef(Link)
      color="primary"
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/details/97074"
    >
      <ForwardRef(Typography)
        variant="subtitle2"
      >
        View Details
      </ForwardRef(Typography)>
    </ForwardRef(Link)>
  </ForwardRef(CardContent)>
</ForwardRef(Card)>
`;

exports[`should test the CarListItemSkeleton Component should render the component 1`] = `
<ForwardRef(Card)
  className="makeStyles-card-1"
  elevation={0}
  raised={false}
  square={true}
  variant="outlined"
>
  <ForwardRef(Skeleton)
    height={75}
    variant="rectangular"
    width={100}
  />
  <ForwardRef(CardContent)
    className="makeStyles-content-2"
  >
    <ForwardRef(Skeleton)
      height={40}
      width={300}
    />
    <ForwardRef(Skeleton) />
    <ForwardRef(Skeleton)
      width={100}
    />
  </ForwardRef(CardContent)>
</ForwardRef(Card)>
`;
